### PR TITLE
[Layout foundations][Batch 1] Rebuild components with layout primitives

### PR DESCRIPTION
**Pull request by laurkim**
### WHY are these changes introduced?

Related to [7565](https://github.com/Shopify/polaris/issues/7565).

### WHAT is this pull request doing?

Rebuilds the following components to use our layout primitives.
- `ChoiceList` #7354
- `Modal` #7340 
- `AccountConnection` #7341 
- `Banner` #7358 

### How to 🎩

[Spin instance](https://admin.web.layout-rebuild-batch-1-b.lo-kim.us.spin.dev/store/shop1)
[Storybook](https://5d559397bae39100201eedc1-cshuopiodd.chromatic.com/)

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by sarahill — APPROVED** [review]
✅

**Review by chazdean — APPROVED** [review]
LGTM! 🚀

**Review by aveline — APPROVED** [review]

File: polaris-react/src/components/Modal/Modal.scss [hunk 1]
```diff
@@ -2,15 +2,6 @@
...
-.BodyWrapper {
-  // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
-  display: flex;
-  // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
-  flex-grow: 1;
-  overflow-x: hidden;
-  -webkit-overflow-scrolling: touch;
-}
-
 .Body {
```
Comment (alex-page) [inline polaris-react/src/components/Modal/Modal.scss:5]: Should the `.Body` be removed now that we replace it with `<Box>`?

Comment (laurkim) [reply polaris-react/src/components/Modal/Modal.scss:5]: The `Body` class is passed to `Scrollable` on [line 172](https://github.com/Shopify/polaris/blob/fe742a16aaf26a005a1e9912bb7a65c43fbc2bf9/polaris-react/src/components/Modal/Modal.tsx#L172).

**Pull request merged**